src/Numerics/Mesh/FTP.jl

```diff
@@ -0,0 +1,201 @@
+using DocStringExtensions
+using KernelAbstractions
+using CUDA
+
+include("FTP_kernels.jl")
+
+"""
+    ftpxv!(grd::DiscontinuousSpectralElementGrid,
+        mesh::Symbol,
+        opts::Symbol,
+        tropts::Bool,
+        vin::AbstractArray{FT},
+        vout::AbstractArray{FT},
+        vin_den::Union{AbstractArray{FT}, Nothing} = nothing,
+    ) where {FT<:AbstractFloat}
+
+This function computes the inner derivatives of a field using the fast
+tensor-product algorithm on a given mesh. Currently, DG mesh (`:m1`) and
+over-integration mesh (`:m2`) are supported. The function provides the 
+building blocks for constructing various terms in the DG formulation and 
+for constructing some diagnostic fields.
+
+# Arguments
+ - `grd`: DiscontinuousSpectralElementGrid
+ - `mesh`: quadrature mesh (currently, `:m1` and `:m2` supported)
```

Comment:
I'm not such a big fan of this. How would you feel about creating a new type, say `InterpolatedDiscontinuousSpectralElementGrid`, that we used for higher-order quadrature? And then use dispatch to determine which path to take.

Reply:
I like the idea of separating the multi-metrics stuff from the current `DiscontinuousSpectralElementGrid`.

Also, after talking to @skandalaCLIMA I gather that this is mainly for use in the continuous Galerkin code, so maybe `InterpolatedContinuousSpectralElementGrid`? Or something similar.